core/src/main/java/org/geysermc/geyser/inventory/click/ClickPlan.java

```diff
@@ -376,7 +382,7 @@ private void reduceCraftingGrid(boolean makeAll) {
         for (int i = 0; i < gridSize; i++) {
             final int slot = i + 1;
             GeyserItemStack item = getItem(slot);
-            if (!item.isEmpty()) {
+            if (!item.isEmpty() && (!handleBookRecipe || item.getJavaId() != session.getItemMappings().getStoredItems().writtenBook().getJavaItem().javaId())) {
```

Comment:
The second comparison can now be `item.asItem() == Items.WRITTEN_BOOK`

Reply:
Did you mean the third comparison? And `!=` instead of `==`?

If the clone recipe is occurring, and then we should only sub items that are not the written book

Reply:
I just committed to make the changes the way Konicai described, if that's not what Camotoy intended I can change it back